From fcbe80d9f5e84f48345515a1765f2a3fa452fc73 Mon Sep 17 00:00:00 2001
From: Sebastian Reichel <sre@debian.org>
Date: Mon, 1 Dec 2014 16:46:05 +0100
Subject: [PATCH] fsogsmd: remove global theServiceDependencies variable

* Global non static variables are no longer supported in
  Vala 0.26 resulting in FTBFS. This patch simply removes
  theServiceDependencies, since it has not been used by
  any plugin.

* MJ: Imported from
  http://anonscm.debian.org/gitweb/?p=pkg-fso/fso-gsmd.git;a=summary

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/lib/modem.vala                     |  3 ---
 src/plugins/dbus_service/resource.vala | 11 -----------
 2 files changed, 14 deletions(-)

diff --git a/src/lib/modem.vala b/src/lib/modem.vala
index 201eeca..447f455 100644
--- a/src/lib/modem.vala
+++ b/src/lib/modem.vala
@@ -23,9 +23,6 @@ namespace FsoGsm
 {
     public const string CONFIG_SECTION = "fsogsm";
     internal const string PPPD_DEFAULT_COMMAND = "/usr/sbin/pppd";
-
-    // This is used to let the plugins define their service dependencies
-    public GLib.List<string> theServiceDependencies = new GLib.List<string>();
 }
 
 /**
diff --git a/src/plugins/dbus_service/resource.vala b/src/plugins/dbus_service/resource.vala
index 9a55e9f..792342b 100644
--- a/src/plugins/dbus_service/resource.vala
+++ b/src/plugins/dbus_service/resource.vala
@@ -63,18 +63,7 @@ public class DBusService.Resource : FsoFramework.AbstractDBusResource
     {
         var dependencies = new GLib.HashTable<string,GLib.Variant?>( GLib.str_hash, GLib.str_equal );
 
-        // Service dependencies can be defined dynamically by the plugins with accessing
-        // the theServiceDependencies global variable.
         string services = "";
-        bool first = true;
-        foreach ( var service in FsoGsm.theServiceDependencies )
-        {
-            if ( !first )
-                services += ",";
-            services += service;
-            first = false;
-        }
-
         dependencies.insert( "services", services );
 
         return dependencies;
-- 
2.1.3

